src/pytorch_lightning/strategies/deepspeed.py

```diff
@@ -564,15 +564,7 @@ def _set_deepspeed_activation_checkpointing(self) -> None:
     def _initialize_deepspeed_inference(self, model: Module) -> None:
         # todo: Currently DeepSpeed requires optimizers at inference to partition weights correctly
```

Comment:
@rohitgr7 I think we need to deal with this TODO before we can make this change. I think that is the real blocker, and the reason why Sean did it the way it is right now.

Reply:
@rohitgr7
Because of the above facts, it seems like it's not possible to disable optimizer initialization for evaluation.
I think of another solution:
Why optimizer initialization works correctly when running `.fit()`, but fail when evaluating? If there's a difference in how the optimizer is initialized, is it possible to initialize it during evaluation the same way as in `.fit()`?

Reply:
ah! should have read the TODO. Let's me investigate more on this :)

Reply:
this is supported. added more details in description.